learn/security/tenant_tokens.md

```diff
@@ -0,0 +1,299 @@
+
+## What is multitenancy?
+
+In software development, multitenancy means that multiple users with different interests—also called tenants—share the same computing resources. Proper multitenancy is crucial in cloud computing services such as [Digital Ocean's Droplets](https://www.digitalocean.com/products/droplets) and [Amazon's AWS](https://aws.amazon.com/).
+
+If your Meilisearch application stores sensitive data like medical records in the same index, it is very important to make sure users can only search through their own documents.
+
+### What are tenant tokens and how do they solve multitenancy in Meilisearch?
+
+Tenant tokens are small packages of encrypted data presenting proof a user can access a certain index. They contain not only security credentials, but also instructions on which documents within that index the user is allowed to see.
+
+To use tokens in Meilisearch, you only need to have a system for token generation in place. The quickest method to generate tenant tokens is [using one of our official SDKs](#generating-tenant-tokens-with-an-sdk). It is also possible to [generate a token from scratch](#generating-tenant-tokens-without-a-meilisearch-sdk).
+
+Tenant tokens do not require you to configure any specific [instance options](/learn/configuration/instance_options.md) or [index settings](/learn/configuration/settings.md). They are also meant to be short-lived—Meilisearch does not store nor keeps track of generated tokens.
+
+## Generating tenant tokens with an SDK
+
+If you are developing an application that allows patients to search through medical records, it is crucial that each user can only see their own records and not anyone else's.
+
+The code in this example imports the SDK, creates a filter based on the current user's ID, and feeds that data into the SDK's `generateTenantToken` function. Once the token is generated, it is stored in the `token` variable:
+
+```js
+// placeholder code sample. Final version will be supplied by the integrations team
+// <CodeSamples id="tenant_token_guide_generate_sdk_1">
+const client = new Meilisearch({host: 'meilisearch'});
+
+const apiKey = '[an_api_key]';
+const currentUserID = '[placeholder_user_id]';
+
+const searchRules = {
+  'patient_medical_records': {
+    'filter': `user_id = ${currentUserID}`
+  }
+};
+
+const expiryDate = 1645461882;
+
+const token = client.generateTenantToken(searchRules, expiryDate, apiKey);
+```
+
+There are three things you should pay attention to when using an SDK to generate a tenant token. In the above example, they are stored in `searchRules`, `apiKey`, and `expiryDate`.
+
+`searchRules` must be an object containing at least one search rule. A search rule must specify either an index name or a `*` wildcard. Search rules must also declare at least one filter. In this example, the filter makes it so a user only has access to their own records.
```

Comment:
I'm not sure about this, unless I did not understand the sentence but these are valid and non valid formats of searchRules: 

✅
```
searchRules: ["*"]
```

❌
```
searchRules: "*"
```
✅
```
searchRules: {}
```

✅
```
searchRules: {
    movies: {}
}
```

✅
```
searchRules: {
    movies: null
}
```

✅
```
searchRules: {
    movies: {
       filters: "id = 1"
    }
}
```

Reply:
I believe the confusion should at least be partly cleared with @dichotommy's suggestion. 

Do let me know otherwise.